system/security/shared/seclib.hpp

```diff
@@ -198,10 +217,60 @@ enum authStatus : int
     AS_PASSWORD_VALID_BUT_EXPIRED = 5//user entered valid password, but authentication failed because it is expired
 };
 
+enum SecUserFeatureIndex : unsigned
+{
+    SUF_GetName,
+    SUF_SetName,
+    SUF_GetFullname,
+    SUF_SetFullName,
+    SUF_GetFirstName,
+    SUF_SetFirstName,
+    SUF_GetLastName,
+    SUF_SetLastName,
+    SUF_GetEmployeeID,
+    SUF_SetEmployeeID,
+    SUF_GetEmployeeNumber,
+    SUF_SetEmployeeNumber,
+    SUF_GetDistinguishedName,
+    SUF_SetDistinguishedName,
+    SUF_GetRealm,
+    SUF_SetRealm,
+    SUF_GetFqdn,
+    SUF_SetFqdn,
+    SUF_GetPeer,
+    SUF_SetPeer,
+    SUF_GetStatus,
+    SUF_SetStatus,
+    SUF_GetAuthenticateStatus,
+    SUF_SetAuthenticateStatus,
+    SUF_Credentials,
+    SUF_GetUserID,
+    SUF_CopyTo,
+    SUF_GetPasswordExpiration,
+    SUF_SetPasswordExpiration,
+    SUF_GetPasswordDaysRemaining,
+    SUF_GetProperty,
+    SUF_SetProperty,
+    SUF_GetPropertyInt,
+    SUF_SetPropertyInt,
+    SUF_GetPropertyIterator,
+    SUF_Clone,
+    SUF_GetDataElement,
+    SUF_GetDataElements,
+
+    SUF_NUMBER_OF_INDICES
```

Comment:
How safe is this for an externally developed security manager plugin` that is not built with the platform?  How would you want the plugin to behave if the platform later adds new SUF features and does not rebuild that plugin?

Reply:
Based on comments from @afishbeck, the enumerations are being redefined to define the bit values, and not indices (with number of indices replaced by a mask for all). As long as the bit values aren't redefined from build to build, a third party plugin should work with the bits it knows about.